concent_api/core/payments/report.py

```diff
@@ -0,0 +1,759 @@
+    payment_info = force_payment_committed_pending_response.payments.order_by('id').last()
+    for settlement_payment in matched_objects:
+        if settlement_payment.closure_time != parse_datetime_to_timestamp(payment_info.payment_ts):
+            errors.append(
+                Error(
+                    matching_error=MatchingError.FORCE_PAYMENT_COMMITED_PAYMENT_TS_DIFFERS_FROM_SETTLEMENT_PAYMENT_CLOSURE_TIMESTAMP,
+                    message=f'PaymentInfo with pk {payment_info.pk} payment ts value {payment_info.payment_ts} '
+                            f'related to stored PendingResponse of type ForcePaymentCommitted '
+                            f'with pk {force_payment_committed_pending_response.pk} '
+                            f'differs from the settlement payment with tx_hash {settlement_payment.tx_hash} '
+                            f'closure time value {settlement_payment.closure_time}.',
+                )
+            )
+
+
+def check_error_more_than_one_subtask_result_of_the_same_type_for_subtask(
+    errors: List[Error],
+    matched_objects: List,
+    additional_verification_payment: CoverAdditionalVerificationEvent,
+) -> None:
+    """ Add error if there is more than two SubtaskResultsSettled or SubtaskResultsRejected for the same subtask. """
+    if (
+        any(
+            [subtask_result.response_type == PendingResponse.ResponseType.SubtaskResultsSettled.name  # pylint: disable=no-member
+             for subtask_result in matched_objects]
+        ) and
+        any(
+            [subtask_result.response_type == PendingResponse.ResponseType.SubtaskResultsRejected.name  # pylint: disable=no-member
+             for subtask_result in matched_objects]
+        )
+    ):
+        errors.append(
+            Error(
+                matching_error=MatchingError.MORE_THAN_ONE_SUBTASK_RESULT_FOR_SUBTASK,
+                message=f'Subtask related with CoverAdditionalVerificationEvent with '
+                        f'tx_hash {additional_verification_payment.tx_hash} '
+                        f'and subtask_id {additional_verification_payment.subtask_id} '
+                        f'has both SubtaskResultsSettled or SubtaskResultsRejected.',
+            )
+        )
+
+
+def check_error_deposit_claim_exists_for_settlement_payment(errors: List[Error], tx_hash: str) -> None:
+    """ Add error if there is DepositClaim related with settlement payment. """
+    if DepositClaim.objects.filter(tx_hash=tx_hash).exists():
+        errors.append(
+            Error(
+                matching_error=MatchingError.DEPOSIT_CLAIM_RELATED_TO_SETTLEMENT_PAYMENT_EXISTS,
+                message=f'DepositClaim still exists for transaction with tx_hash {tx_hash}',
+            )
+        )
+
+
+def check_error_deposit_claim_exists_for_forced_payment(errors: List[Error], tx_hash: str) -> None:
+    """ Add error if there is DepositClaim related with forced payment. """
+    if DepositClaim.objects.filter(tx_hash=tx_hash).exists():
```

Comment:
This function and the function above check the same thing and only have different messages. Just like in my previous comment, this seems incomplete. The check is not self-contained so it's hard to even say if your check here is right or wrong. It sometimes is and sometimes is not, depending on how you use the function. It makes it harder to understand and review this.